src/main/logic/Bullets.h

```diff
@@ -12,6 +12,11 @@
 
 namespace gs {
 
+class Bullets;
```

Comment:
Currently I am using the typedef inside the classes just to tidy a few things up. If you prefer I can remove the forward declaration, move the typedef below and write out BulletsShPtr in long hand during the class. It is up to you, I don't know which is the more common approach.